### ISSUE TITLE
name 'self' is not defined

### ISSUE THREAD

**Issue by curtis90** [ManuSetty/wishbone#9]
Hi, 

After installing wishbone following the tutorial, I got an error as "NameError: name 'self' is not defined".

I use Ubuntu 16.04 desktop in Virtualbox. Everything is Ok, but I can not run the program.

Thx!

**Comment by ManuSetty** [comment]
Can you please paste the full Error traceback? That will help a lot to diagnose the error.

**Comment by curtis90** [comment]
Linux
Traceback (most recent call last):
  File "./wishbone_gui.py", line 473, in <module>
    launch()
  File "./wishbone_gui.py", line 464, in launch
    self.lift()
NameError: name 'self' is not defined

**Comment by curtis90** [comment]
yuancao@yuancao:~/wishbone$ sudo pip3 install .
[sudo] password for yuancao:
The directory '/home/yuancao/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/yuancao/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Processing /home/yuancao/wishbone
Requirement already satisfied (use --upgrade to upgrade): wishbone==0.2.2 from file:///home/yuancao/wishbone in /usr/local/lib/python3.5/dist-packages
Requirement already satisfied (use --upgrade to upgrade): numpy>=1.10.0 in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): pandas>=0.18.0 in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): scipy>=0.14.0 in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): tsne==0.1.3 in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): matplotlib in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): seaborn in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): sklearn in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): networkx in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): fcsparser in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): statsmodels in /usr/local/lib/python3.5/dist-packages (from wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): python-dateutil>=2 in /usr/local/lib/python3.5/dist-packages (from pandas>=0.18.0->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): pytz>=2011k in /usr/local/lib/python3.5/dist-packages (from pandas>=0.18.0->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): Cython>=0.19.1 in /usr/local/lib/python3.5/dist-packages (from tsne==0.1.3->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): cycler in /usr/local/lib/python3.5/dist-packages (from matplotlib->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): pyparsing!=2.0.4,>=1.5.6 in /usr/local/lib/python3.5/dist-packages (from matplotlib->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): scikit-learn in /usr/local/lib/python3.5/dist-packages (from sklearn->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): decorator>=3.4.0 in /usr/local/lib/python3.5/dist-packages (from networkx->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python3/dist-packages (from fcsparser->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): patsy in /usr/local/lib/python3.5/dist-packages (from statsmodels->wishbone==0.2.2)
Requirement already satisfied (use --upgrade to upgrade): six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2->pandas>=0.18.0->wishbone==0.2.2)


________________________________
发件人: ManuSetty <notifications@github.com>
发送时间: 2016年6月15日 4:22:16
收件人: ManuSetty/wishbone
抄送: curtis90; Author
主题: Re: [ManuSetty/wishbone] name 'self' is not defined (#9)


Can you please paste the full Error traceback? That will help a lot to diagnose the error.

―

**Comment by ManuSetty** [comment]
Sorry, the GUI has been developed and tested for Windows and MAC. We are working on a Linux version now. The command line interface as detailed in this [notebook](http://nbviewer.jupyter.org/github/ManuSetty/wishbone/blob/master/notebooks/Wishbone_for_mass_cytometry.ipynb) will however work on Linux as well. 

I will leave the issue open and notify when the Linux version is ready.

**Comment by curtis90** [comment]
Thank you.  BTW, how to install it on Windows?


Best,


Yuan

________________________________
发件人: ManuSetty <notifications@github.com>
发送时间: 2016年6月15日 5:12:01
收件人: ManuSetty/wishbone
抄送: curtis90; Author
主题: Re: [ManuSetty/wishbone] name 'self' is not defined (#9)


Sorry, the GUI has been developed and tested for Windows and MAC. We are working on a Linux version now. The command line interface as detailed in this notebook<http://nbviewer.jupyter.org/github/ManuSetty/wishbone/blob/master/notebooks/Wishbone_for_mass_cytometry.ipynb> will however work on Linux as well.

I will leave the issue open and notify when the Linux version is ready.

―

**Comment by ManuSetty** [comment]
The commands listed in README should work for Windows as well

```
    $> git clone git://github.com/manusetty/wishbone.git
    $> cd wishbone
    $> pip3 install .
```

**Comment by ManuSetty** [comment]
This is fixed in v0.2.3. Please use the latest version of Wishbone and let us know if run into any other issues.
